R/eval_forecasts.R

```diff
@@ -284,26 +356,71 @@ eval_forecasts <- function(data,
   dat[, names(dat)[grepl("sampl_", names(dat))] := NULL]
   dat[, c("sharpness", "bias", "dss", "crps") := NULL]
 
-
   # merge with previous data
   merge_cols = colnames(dat)[!colnames(dat) %in% c("pit_p_val", "pit_sd")]
   res <- merge(data, dat, by = merge_cols)
 
-  if (prediction_type == "continuous") {
-    scores <- c("sharpness", "bias", "dss", "crps", "log_score", "pit_p_val")
-  } else {
-    scores <- c("sharpness", "bias", "dss", "crps", "pit_p_val", "pit_sd")
-  }
-
-  # aggregate scores so there is only one score per observed value
-  res <- res[, lapply(.SD, mean, na.rm = TRUE),
-             .SDcols = scores,
+  # @ SAM
+  # scores are computed over 'by'. This means that scores are duplicated.
+  # we could either leave it unsummarised - or filter out all duplicates, so that only
+  # unique scores remain. Do we want that?
+  # previous version was mean, na.rm = TRUE, but unqique should do the same
+  res <- res[, lapply(.SD, unique),
+             .SDcols = colnames(res) %like% "pit_|bias|sharpness|dss|crps|log_score",
              by = c("id", by)]
 
+
   if (summarised) {
+    # add quantiles
+    if (!is.null(quantiles)) {
+      res[, paste0("crps_", quantiles) := as.list(quantile(crps,
```

Comment:
as this is the same code repeated it would be nice if it was an internal function but makes no functional difference.

Reply:
good point :) that is now done